units/src/result.rs

```diff
@@ -215,6 +215,92 @@ impl<T: fmt::Debug> NumOpResult<T> {
     pub fn is_error(&self) -> bool { !self.is_valid() }
 }
 
+// Implement Add/Sub on NumOpResults for all wrapped types that already implement Add/Sub on themselves
```

Comment:
Ha, interesting puzzle working out the 'on themselves' bit and how the compiler doesn't get itself in knots because of the `AddAsign<Rhs = Self>` - that was fun to review.